---
title: Resolving gut microbiome networks within Chiropterans
authors:
  - name: Timothy J. Rogers
    department: Department of Bioinformatics
    affiliation: University of North Carolina
    location: Charlotte, NC 28223
    email: troger50@charlotte.edu
  - name: Laurel R. Yohe
    department: Department of Bioinformatics and Genomics
    affiliation: North Carolina Research Center (NCRC)
    location:  Kannapolis, NC 28081
    email:  lyohe1@charlotte.edu
  - name: Richard A. White ${^\dagger}$
    department: Department of Bioinformatics
    affiliation: University of North Carolina
    location: Charlotte, NC 28223
    email: rwhit101@charlotte.edu
abstract: |
  Enter the text of your abstract here.
keywords:
  - Chiropterans
  - microbiome
  - virome
  - metagenomics
bibliography: "`r rbbt::bbt_write_bib('references.bib', overwrite = TRUE)`" 
#bibliography: references.bib
biblio-style: unsrt
output: rticles::arxiv_article
geometry: margin=1.0in
font-size: 11pt
header-include:
  -\usepapackage(helvet)
  -\renewcommand*\familydefault{\sfdefault}

---
${\dagger}$ To whom correspondence should be addressed

\newpage
```{r}
library(data.table)
library(tidyverse)
```
## Introduction
* Bats are known carriers of human associated pathogens
* The reason bats are bioreactors is not understood
* The diet of bats may contribute to the gut microbiota makeup
* Phage associated with these microbiota can benifit and hinder microbial populations and have an impack on bat immune responses
* Some ideas suggest viral tolerance is linked to 
  - Uniqueness of bats and their variation in Diets
    - Diversity
    - Ecological role
    - Pathogenic role
* Methods for characterizing microbiome and virome. As well as methods for linking the two  
  - Culture dependant vs independent
    - What has been discovered with these methods
    - How these methods have been applied to bats 
      - What has been found
      - What has yet to be discribed
- Goal of this study

## Materials and Methods
* Sample location discription
* Sample collection
* Sample processing
### Phase genomics portion
* (AND) 

### In house analysis of MAGs, vcontigs, and viral-host pairing
#### vOTU curration
* (AND) Genomad (v x.x.x) was used to varify contigs identified as viral by phase genomics and for taxonomic identification. 
* (AND) We also ran all assembly files from phase genome through genomad to identify potential viral sequences that the phase genome pipeline may have missed
* (AND) Quality filtered viral sequences were then clustered into species-level equivalent viral operational taxanomic units (vOTUs) at 95% average nucleatide identity over 85% of the alignment fraction of the shorter sequence using the greedy centroid algorithm (anicalc.py and aniclust.py) from checkV (v x.x.x).

#### MAG curration
* (AND) MAGs from phase metagenomic were quality filtered using checkM (v x.x.x).
* (AND) Quality filtered MAGs were then dereplicated using dRep (v x.x.x) at 99% ANI with the following settings (--S_algorithm fastANI, -comp 50, --SkipMash)
* (AND) MAGs representatives were taxonomically identified using the Genome Taxonomy Database Toolkit (GTDB-Tk v x.x.x)

#### vOTU and MAG coverage
* (AND) Quality controled reads were mapped to the vOTUs and representative MAGs using bowtie (v x.x.x) and sam files were converted to indexed and sorted bam files using samtools (v x.x.x).
* (AND) Bam files were then fed into anvi'o (v x.x.x) to calculate the Q2Q3 coverage of vOTUs and MAGs across bat species.
* (AND) Q2Q3 coverages for both vOTUs and MAGs were then normalized by sequence length.
* (AND) For alpha diveristy analyzes, normalized Q2Q3 coverages for both vOTUs and MAGs were rarefied using the rrarify function of the vengan  R package.
* (AND) For beta diversity analyzes, normalized Q2Q3 coverages for both vOTUs and MAGs were used to create rarefied Hellinger distance matrices using the vegan (v x.x.x) and labdsv (v x.x.x) R packages.
* (AND) For abundance pattern analyzes, normalized Q2Q3 coverages were coverted into units of GCPM (Genome Copy Per Million Reads) for MAGs, as laid out in Rogers et al. 2022 (citation), and TPM (Transcripts Per Million Reads) for vOTUs.
* (AND) Abundance correlations of vOTUs and MAGs across and within samples were determined using hierarchical clustering of Spearman rank correlation distance matrice in base R (v x.x.x).

#### Metabolic predictions
* The program metacerberus (v x.x.x) was used for gene anotation of both viral sequences and MAGs using the following data bases: Functional Ontology Assignments for Metagenomes (FOAM), KEGG, CAZy/dbCAN, VOG, pVOG, PHROG, and COG.

#### AMR and CAZY gene abundance patterns
* (AND) Read coverage and abundance for AMR and CAZY genes were calculated using a custom ihhouse script.

#### Linking virus to host
* (AND) IPHoP (v x.x.x) was used to link all viral sequences to all medium to high quality MAGs, not just to the representative vOTUs and MAGs. 
* (AND) The viral and MAG IDs of those within the viral-host predictions from both IPHoP and Phase genomes were replaced with the corresponding representative vOTU and MAG IDs.

#### Statistical analysis
* (AND) Differences in MAG and vOTU composition among bat species were analyzed via distance based redundancy analysis (db-RDA) on a quntitative Hellinger distance matirx. 
* (AND) The richness and evenness of MAGs and vOTUs were compared using the R package microbiome for richness (chao1) and the package hillR (v x.x.x) for evenness (Pielou's index).
* (AND) Statistical significants was determined based on 9999 permutations of the data in the vegan R package.

## Results
* MAG
  - Number and quality of MAGs
  - Taxonomic idenification
  - Distribution patterns
    - Alpha Diversity
      - Richness
      - Evenness
    - Beta Diversity
    - Metabolic profiles of gut microbiome across bat species

* vOTUs
  - Number of vOTUs
  - Quality of vOTUs
  - Taxonomic idenification
  - Distribution patterns
    - Alpha Diversity 
      - Richness
      - Evenness
    - Beta Diversity
    - AMGs and viral vectors

* Virus-host matching
  - IPHop
    - CRISPR
    - tetranucleotide frequency
    - blast
  - HiC

### MAGs
```{r, echo=FLASE}
# Quality and length of rarefied MAGs
mag_stats<-fread("/projects/raw_lab/projects/Bats/bat_gut_micro/phasegenome_out/data/results/tables/mag_stats.csv")
# Abudance information:
mag_fam_rel_abun<-fread("/projects/raw_lab/projects/Bats/bat_gut_micro/phasegenome_out/data/results/tables/mag_fam_rel_abun.csv")
phylum_rel_abun<-mag_fam_rel_abun%>%select(-V1,-class:-family)%>%
  aggregate(.~phylum,sum)
colSums(phylum_rel_abun[,-1])
```
* (AND) We recovered `r mag_stats[domain=="Archaea",.N]` Archaeal and `r mag_stats[mag_stats$domain=="Bacteria",.N]` Bacterial medium to high quality MAGs with $\geq$ `r paste0(round(min(mag_stats$completeness)),"%")` and $\lt$ `r paste0(round(max(mag_stats$contamination)),"%")` redundancy.
* (AND) Of these `r nrow(mag_stats)` MAGs, `r length(mag_stats$genome[mag_stats$completeness>=90 & mag_stats$contamination<10,])` were $\geq$ 90% complete and $\lt$ 10% redundant, while YY are of high quality as determined by parameters layed out in XXX et al. (** TABLE **).
* (AND) Bacillota, Actinomycetota, and Pseudomonadata were the predominate phylum across the gut microbiome within all 3 of the bat species.
* (AND) No significant differnces were found in the richness (Choa1) or evenness (Pielou's Index) of prokaryotic MAGs across the 3 bat species.
* (AND) db-RDA analysis revealed no significant differnce in the $B$-diversity of the prokaryotic MAGs across the 3 bat species.
\includegraphics{../data/results/figures/FigX_top10_most_abun_viral_taxa.pdf}
### vOTUs
```{r,echo=FALSE}

```
- A total of XX vOTUs were recovered.
- X, Y, and Z were the most predominate viral taxonomic unit across the gut virome with all 3 of the bat species.

#### Metabolic capabilities
* (AND)
## Descusion

## Conclusion

## Acknowledgement
